temporalio/runtime.py

```diff
@@ -124,15 +130,100 @@ def _to_bridge_config(self) -> temporalio.bridge.runtime.LoggingConfig:
             filter=self.filter
             if isinstance(self.filter, str)
             else self.filter.formatted(),
-            # Log forwarding not currently supported in Python
-            forward=False,
+            forward_to=None if not self.forwarding else self.forwarding._on_logs,
         )
 
 
 LoggingConfig.default = LoggingConfig(
     filter=TelemetryFilter(core_level="WARN", other_level="ERROR")
 )
 
+_module_start_time = time.time()
+
+
+@dataclass
+class LogForwardingConfig:
+    """Configuration for log forwarding from Core.
+
+    Configuring this will send logs from Core to the given Python logger. By
+    default, log timestamps are overwritten and internally throttled/buffered
+    for a few milliseconds to prevent overloading Python. This means those log
+    records may have a time in the past and technically may appear out of order
+    with Python-originated log messages by a few milliseconds.
+
+    If for some reason lots of logs occur within the buffered time (i.e.
+    thousands), they may be sent earlier. Users are discouraged from using this
+    with ``TRACE`` Core logging.
+
+    All log records produced have a ``temporal_log`` attribute that contains a
+    representation of the Core log. This representation has a ``fields``
+    attribute which has arbitrary extra data from Core. By default a string
+    representation of this extra ``fields`` attribute is appended to the
+    message.
+    """
+
+    logger: logging.Logger
+    """Core logger messages will be sent to this logger."""
```

Comment:
One thing here, you might want to consider emitting to a logger with the module name. That's fairly standard Python practice.

Reply:
Users choose the logger to provide, if they want to provide a logger with whatever name they want. I also append the core log target to give logger name by default, but I don't want to be in charge of creating and managing loggers.

Reply:
This is very non Pythonic practice based on my experience.
I'd at minimum mark this API experimental.

Reply:
I don't think it's non-Pythonic to accept a user-configured logger from a user configured for external logging. It's bad to assume that this special case of using an logger may not want to be configured in a custom way from other logging. It's not like we do this for our normal logs, this is a very special case and they may want to customize the logger for this case.

Can you clarify exactly what you think should change? Should we use the global default logger factory? Should we continue appending the core target to the name of the log records? You think users should be able to explicitly set handlers for core logging?

I think you need better justifications and clarifications here. You can't assume your experience in normal logging situations applies to pushed logs in this special circumstance.

Reply:
My suggestion is that you emit to a logger that you create (e.g. `logging.getLogger(__name__)`) and let users handle logs as they normally would with a global logging configuration.
That way, all they need to do is opt-in to forwarding, and the rest "just works".

Reply:
The difference between `forwarding = True` and `logger = logging.getLogger(__name__)` is negligible for "just works".

Now, where is this logger stored? How do I add a handler to that logger? How do I have different runtimes have different loggers? Should I append the Core target to that name on each log record? Is the only reason you're wanting to do this is because `True` appears easier than `logging.getLogger(__name__)` from a user POV regardless of all of the known negatives?